<a href="https://colab.research.google.com/github/naveenkumark1/EVA4/blob/master/Week1/python101.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
print("OM")

OM


This python notbook is for the python 101 practice code

This is a MUST Python Refresher. You must finish this notebook as a part of overall grading.

Majority of this notebook is inspired from: Stanford's Python Refresher which can be found at: https://cs231n.github.io/python-numpy-tutorial/

We have modified the original notebook and added some questions here. You need to find the solutions and provide answers in your Q1 - Python 101 Quiz.

# Python
Python is a high-level, dynamically typed multiparadigm programming language. Python code is often said to be almost like pseudocode, since it allows you to express very powerful ideas in very few lines of code while being very readable.
New Section

# Python Versions
There are currently two different supported versions of Python, 2.7 and 3.5. Somewhat confusingly, Python 3.0 introduced many backwards-incompatible changes to the language, so code written for 2.7 may not work under 3.5 and vice versa. For this class all code will use Python 3.5 or higher.

You can check your Python version at the command line by running.

In [3]:
!python --version

Python 3.6.9


# Basic data types

Like most languages, Python has a number of basic types including integers, floats, booleans, and strings. These data types behave in ways that are familiar from other programming languages.

Numbers: Integers and floats work as you would expect from other languages:

In [4]:
x = 37 # Initial Value of x
print(type(x)) # Prints "<class 'int'>"
print(x)       # Prints "3"
print(x + 1)   # Addition; prints "4"
print(x - 1)   # Subtraction; prints "2"
print(x * 2)   # Multiplication; prints "6" - Error
print(x ** 2)  # Exponentiation; prints "9" - Error
x += 1
print(x)  # Prints "4" - Error
x *= 2
print(x)  # Prints "8" - Error
y = 2.5
print(type(y)) # Prints "<class 'float'>"
print(y, y + 1, y * 2, y ** 2) # Prints "2.5 3.5 5.0 6.25"

x = x * y + x + y # Final value of x
print("Final Value of x is:", x)

<class 'int'>
37
38
36
74
1369
38
76
<class 'float'>
2.5 3.5 5.0 6.25
Final Value of x is: 268.5


# Quiz Question 1

With what integer value shall x be initialized as, such that Final Value of x is 268.5? Please mention the answer in INT value without decimal (e.g. 4, 2, 12, etc.).

Please answer this question in the Quiz Section.

# Booleans

Python implements all of the usual operators for Boolean logic, but uses English words rather than symbols (&&, ||, etc.):

In [5]:
t = True
f = False
print(type(t)) # Prints "<class 'bool'>"
print(t and f) # Logical AND; prints "False"
print(t or f)  # Logical OR; prints "True"
print(not t)   # Logical NOT; prints "False"
print(t != f)  # Logical XOR; prints "True"

<class 'bool'>
False
True
False
True


# Quiz Question 2

What is the value of

In [6]:
not ( not t + f * t)

True

# Strings

Python has great support for strings:

In [7]:
hello = 'hello'    # String literals can use single quotes
world = "world"    # or double quotes; it does not matter.
print(hello)       # Prints "hello"
print(len(hello))  # String length; prints "5"
hw = hello + ' ' + world  # String concatenation
print(hw)  # prints "hello world"
hw6 = '%s %s %d' % (hello, world, 6)  # sprintf style string formatting
print(hw6)  # prints "hello world 6"

location = 0
print(hw6[location])

hello
5
hello world
hello world 6
h
